ariadne/cardano/src/Ariadne/Wallet/Backend.hs

```diff
@@ -21,68 +22,73 @@ import Ariadne.Wallet.Backend.KeyStorage
 import Ariadne.Wallet.Backend.Restore
 import Ariadne.Wallet.Backend.Tx
 import Ariadne.Wallet.Cardano.Kernel.DB.AcidState (Snapshot(..), defDB)
-import Ariadne.Wallet.Cardano.WalletLayer.Kernel (bracketPassiveWallet)
+import Ariadne.Wallet.Cardano.WalletLayer.Kernel (passiveWalletLayerComponent)
 import Ariadne.Wallet.Cardano.WalletLayer.Types (PassiveWalletLayer(..))
 import Ariadne.Wallet.Face
 
+-- | This is what we create initially, before actually creating 'WalletFace'.
+data WalletPreface = WalletPreface
+    { wpBListener :: !BListenerHandle
+    , wpAddUserSecret :: !(TVar UserSecret -> IO ())
+    , wpMakeWallet :: !(CardanoFace -> ((Doc -> IO ()) -> WalletFace, IO ()))
+    }
 
-createWalletBackend :: WalletConfig -> (WalletEvent -> IO ()) -> ComponentM
-  ( BListenerHandle
-  , TVar UserSecret -> IO ()
-  , CardanoFace ->
-    ((Doc -> IO ()) -> WalletFace, IO ())
-  )
-createWalletBackend walletConfig sendWalletEvent = buildComponent_ "Wallet" $ do
-  walletSelRef <- newIORef Nothing
-  -- TODO: Do I need to close session on exit?
-  acidDb <- openLocalStateFrom walletAcidDbPathPlaceholder defDB
+createWalletBackend ::
+       WalletConfig -> (WalletEvent -> IO ()) -> ComponentM WalletPreface
```

Comment:
Indentation is weird.

Reply:
No idea why `hindent` does it this way :man_shrugging: